Update the Readme and add PR template (#18)
dotnet/spa-templates: README.md

> This repo is home for the SPA (Single Page Application) project templates which ship as part of ASP.NET Core.

Reviewer: ```suggestion
This repo is home for the Single Page Application (SPA) project templates which ship as part of ASP.NET Core and the .NET SDK.
```

At the moment, those are `Angular` and `React` templates.

Besides these two templates, we also want this repo to serve as a knowledge-base for all community-owned SPA project templates out there.

> We expect to maintain that list in the `SpaProjectTempltes.md` file. If you know of any such project templates, which are popular, send us a PR with added information in that file, and we'll include it, it it meets the popularity bar.

Reviewer: ```suggestion
We expect to maintain that list in the `SpaProjectTempltes.md` file. If you know of any such project templates, which are popular, send us a PR with added information in that file, and we'll include it, if it meets the popularity bar.
```

Reviewer: Actually, I don't think we should have a popularity bar. I care more about having a basic quality bar: HTTPS enabled, good dev experience, publish works as expected, JS framework code kept up to date. The Awesome Blazor repo uses some tags to indicate freshness that might be useful here.

Author: Thanks Dan. I don't want us to be in position to validate all those with every PR submission. I'd rather let the community to share feedback about how healthy this list is.
That said, I'm ok simply listing those requirements but accepting any submissions with a quick scan of the code and the repo / package. I'll then let the community react in case we made a mistake and have included something not worthy.

This repo is home for the Single Page Application (SPA) project templates which ship as part of ASP.NET Core and the .NET SDK.
At the moment, those are `Angular` and `React` templates.

Besides these two templates, we also want this repo to serve as a knowledge-base for all community-owned SPA project templates out there.

> We expect to maintain that list in the `SpaProjectTempltes.md` file. If you know of any such project templates, which will meet some basic quality bar, submit a PR and we'll consider it. Things that we'll be looking for are:

Reviewer: Instead of maintaining the community template list in a separate file, can we just list them in the readme? One less click.

This repo is home for the SPA (Single Page Application) project templates which ship as part of ASP.NET Core.
At the moment, those are `Angular` and `React` templates.

Besides these two templates, we also want this repo to serve as a knowledge-base for all community-owned SPA project templates out there.

> We expect to maintain that list in the `spaProjectTempltes.md` file. If you know of any such project templates, which are popular, send us a PR with added information in that file, and we'll include it, it it meets the popularity bar.

Reviewer 2: nit: Capitalize S in `spaProjectTemplates.md`

Author: Actually, the file name is in lowercase, so I'd rather keep this as is.

We try to keep the list of community owned SPA project templates up to date as well as want to make sure it is a good quality list for everyone to refer to.
To keep the quality high, we have some requirements each proposed project template should comply to:

> - The project templates hould be available as a NuGet package

Reviewer 2: Instead of trying to specify the exact criteria, maybe we should just say that submissions will be evaluated on a case-by-case basis to avoid adding templates that aren't of high-quality and widely enough used.

> This repo is home for the Single Page Application (SPA) project templates which ship as part of ASP.NET Core and the .NET SDK.
> At the moment, those are `Angular` and `React` templates.

Reviewer: ```suggestion
# ASP.NET Core Single Page App templates

This repo is home for the ASP.NET Core Single Page Application (SPA) project templates that ship as part of the .NET SDK. Currently, the .NET SDK includes template for Angular and React.
```

> Besides these two templates, we also want this repo to serve as a knowledge-base for all community-owned SPA project templates out there.
> We expect to maintain that list in the bottom of this file. If you know of any such project templates, which will meet some basic quality bar, submit a PR and we'll consider it. Things that we'll be looking for are:
> - HTTPS enabled by default
> - good dev experience - easy to understand and work with out of the box
> - publishing newly created project works
> - JS framework code kept up-to-date

Reviewer: ```suggestion
The templates in this repo establish a pattern that can be replicated for using ASP.NET Core with other frontend JavaScript frameworks:

- An ASP.NET Core project with a *ClientApp* folder containing the SPA client code.
- Scripts for setting up HTTPS during development using the ASP.NET Core HTTPS development certificate.
- Configuration for the frontend development server to proxy API requests to the ASP.NET Core backend.
- Enable the ASP.NET Core SPA proxy middleware to redirect to the frontend development server when it's running.
- MSBuild logic for running the app during the development and publishing the client app with the ASP.NET Core backend as a single app.
- (Optional) Setup authentication

Check out the [ASP.NET Core docs](https://aka.ms/aspnetcorespa) for additional details on integrating ASP.NET Core with frontend JavaScript frameworks.
```

Placeholder doc link until we get content published.

# ASP.NET Core Single Page App templates

> This repo is home for the ASP.NET Core Single Page Application (SPA) project templates that ship as part of the .NET SDK. Currently, the .NET SDK includes template for Angular and React.

Reviewer: ```suggestion
This repo is home for the ASP.NET Core Single Page Application (SPA) project templates that ship as part of the .NET SDK for Angular and React, as well as a listing of community provided templates for using ASP.NET Core with other frontend JavaScript frameworks.
```

Besides these two templates, we also want this repo to serve as a knowledge-base for all community-owned SPA project templates out there.
We expect to maintain that list in the bottom of this file. If you know of any such project templates, which will meet some basic quality bar, submit a PR and we'll consider it. Things that we'll be looking for are:
- HTTPS enabled by default
- good dev experience - easy to understand and work with out of the box
- publishing newly created project works
- JS framework code kept up-to-date

> We try to keep the list of community owned SPA project templates up to date as well as want to make sure it is a good quality list for everyone to refer to. For that, submissions will be evaluated by our team on case-by-case basis.
> 
> ## How to propose a new SPA project template reference
> 
> We've created a new PR template, to help with new project template proposals.
> The format of each project template in the file should follow existing pattern, which goes as follows:
> 
> ```
> 1. **Template name** / [GitHub](https://github.com/**owner/repo**) /  [NuGet](https://www.nuget.org/packages/**packageName**/)<br />
>   Short description of the template
> ```
> 
> Simply copy the above template to the bottom of this file, and then modify it as necessary.
> 
> When you're ready to propose your changes prepared in your fork, use the `/.github/PULL_REQUEST_TEMPLATE/projectTemplate.md` PR template when submitting a new PR. To do so, append the additional `&template=projectTemplate.md` parameter to the PR creation page link, and you'll get all the info we need to fill in for the PR.
> 

Reviewer: ```suggestion
We try to keep the list of community owned SPA project templates up to date as well as want to make sure it is a good quality list for everyone to refer to. For that, submissions will be evaluated by our team on case-by-case basis.

## How to propose a new SPA project template reference

We've created a new PR template, to help with new project template proposals.
The format of each project template in the file should follow existing pattern, which goes as follows:

```
1. **Template name** / [GitHub](https://github.com/**owner/repo**) /  [NuGet](https://www.nuget.org/packages/**packageName**/)<br />
  Short description of the template
```

Simply copy the above template to the bottom of this file, and then modify it as necessary.

When you're ready to propose your changes prepared in your fork, use the `/.github/PULL_REQUEST_TEMPLATE/projectTemplate.md` PR template when submitting a new PR. To do so, append the additional `&template=projectTemplate.md` parameter to the PR creation page link, and you'll get all the info we need to fill in for the PR.

```

> ## Community owned ASP.NET Core SPA project templates
> The below list represents project template packages on NuGet, which enable different scenarios for ASP.NET Core backed SPA project templates.

Reviewer: ```suggestion
## Community-owned ASP.NET Core SPA project templates

The list below contains community-owned project templates for using ASP.NET Core with various frontend JavaScript frameworks. 
```

<!--
Please use this template for each submission

```

> 1. **Template name** / [GitHub](https://github.com/**owner/repo**) /  [NuGet](https://www.nuget.org/packages/**packageName**/)<br />
>   Short description of the template

Reviewer: ```suggestion
1. **Template name** / [GitHub](https://github.com/**owner/repo**) /  [NuGet](https://www.nuget.org/packages/**packageName**/) - Short description of the template
```
